x/delayedack/eibc.go

```diff
@@ -18,6 +19,34 @@ const (
 	PFMMemoObjectName  = "forward"
 )
 
+var (
+	ErrMemoUnmarshal = fmt.Errorf("unmarshal memo")
+	ErrMemoIsPFM     = fmt.Errorf("EIBC packet with PFM is currently not supported")
+	ErrMemoEibcEmpty = fmt.Errorf("memo IBC field is missing")
+)
+
+func parsePacketMetadata(input string) (*types.PacketMetadata, error) {
+	memo := make(map[string]any)
+	bz := []byte(input)
+	err := json.Unmarshal(bz, &memo)
+	if err != nil {
+		return nil, ErrMemoUnmarshal
+	}
+	if memo[eibcMemoObjectName] == nil {
+		return nil, ErrMemoEibcEmpty
+	}
+	if memo[PFMMemoObjectName] != nil {
+		// Currently not supporting eibc with PFM: https://github.com/dymensionxyz/dymension/issues/599
+		return nil, ErrMemoIsPFM
+	}
+	var metadata types.PacketMetadata
+	err = json.Unmarshal(bz, &metadata)
+	if err != nil {
+		return nil, ErrMemoUnmarshal
```

Comment:
I think there are 2 cases which should also solve the second issue:
1. here when we fail to parse eIBC , return a different error, i.e `ErrEIBCPacketUnmarshal` - this is in order to return an err ack, as in the current situation, failing to unmarshal eibc (i.e there is an eibc memo, it's just malformed), we return nil, which means the IBC tx will go through anyway

2. on line 102, failing to validate, shoudl also return err and prevent from IBC tx to pass.